refactor(linter_codegen): simplify sorting rules

overlookmotel opened the pull request:
Follow-on after #13138, as per [this comment](https://github.com/oxc-project/oxc/pull/13138#discussion_r2328541818).

Pure refactor. Simplify sorting rules, by leaning on derived traits instead of manual implementation.

camc314 commented:
### Merge activity

* **Sep 8, 3:19 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(linter_codegen): simplify sorting rules (#13575)

Follow-on after #13138, as per [this comment](https://github.com/oxc-project/oxc/pull/13138#discussion_r2328541818).

Pure refactor. Simplify sorting rules, by leaning on derived traits instead of manual implementation.
